tdigest/accumulation_tree.py

```diff
@@ -0,0 +1,151 @@
+import operator
+from bintrees.rbtree import RBTree, Node
+
+
+class AccumulationTree(RBTree):
+    def __init__(self, mapper, reducer=operator.add, zero=0):
+        RBTree.__init__(self)
+        self._zero = zero
+        self._mapper = mapper
+        self._reducer = reducer
+
+        self._dirty_nodes = set()
+
+        class AccumulatingNode(Node):
+            """Internal object, represents a tree node."""
+            __slots__ = ['accumulation']
+
+            def __init__(node, key, value):
+                Node.__init__(node, key, value)
+                node.accumulation = self._mapper(value)
+
+            def __setitem__(node, key, value):
+                Node.__setitem__(node, key, value)
+                self._dirty_nodes.add(node.key)
+
+        self._Node = AccumulatingNode
+
+
+    def _new_node(self, key, value):
+        """Create a new tree node."""
+        self._count += 1
+        node = self._Node(key, value)
+        return node
+
+    def insert(self, key, value):
+        try:
+            RBTree.insert(self, key, value)
+        finally:
+            self._update_dirty_nodes()
+
+    def remove(self, key):
+        try:
+            RBTree.remove(self, key)
+        finally:
+            self._update_dirty_nodes()
+
+    def _get_full_accumulation(self, node):
+        if not node:
+            return self._zero
+        else:
+            return node.accumulation
+
+    def _get_right_accumulation(self, node, lower):
+        if not node:
+            return self._zero
+        if(lower <= node.key):
+            return self._reducer(
+                self._reducer(
+                    self._get_right_accumulation(node[0], lower),
+                    self._get_full_accumulation(node[1]),
+                ),
+                self._mapper(node.value),
+            )
+        else:
+            return self._get_right_accumulation(node[1], lower)
+
+    def _get_left_accumulation(self, node, upper):
+        if not node:
+            return self._zero
+        if(upper > node.key):
+            return self._reducer(
+                self._reducer(
+                    self._get_full_accumulation(node[0]),
+                    self._get_left_accumulation(node[1], upper),
+                ),
+                self._mapper(node.value),
```

Comment:
logic like this: https://github.com/tkluck/tdigest/blob/e0acbee7ac74bddb51f8eaa90186d9d7839d6bbf/tdigest/accumulation_tree.py#L132-L137

This looks like we are adding the value to `accumulate`